test/doc/websocket_2_handshaking.cpp

```diff
@@ -132,6 +132,94 @@ snippets()
 
     //]
     }
+
+    {
+        //[code_websocket_2_6
+
+        // a function to select the most preferred protocol from a comma-separated list
+        auto best_protocol = [](string_view offered_tokens) -> std::string
+        {
+            // tokenize the Sec-Websocket-Protocol header offered by the client
+            http::token_list offered( offered_tokens );
+
+            // an array of protocols supported by this server
+            // in descending order of preference
+            static const std::array<string_view, 3>
+                supported = {
+                "v3.my.chat",
+                "v2.my.chat",
+                "v1.my.chat"
+            };
+
+            std::string result;
+
+            for (auto proto : supported)
+            {
+                auto iter = std::find(offered.begin(), offered.end(), proto);
```

Comment:
If subprotocol matching is case-insensitive you could use this function:
https://github.com/boostorg/beast/blob/develop/include/boost/beast/http/rfc7230.hpp#L280

You have to find out if case matters though

Reply:
it's absolutely down to the implementor of the subprotocol